internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -165,11 +163,7 @@ func (r *BundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	if bundle.Status.ObservedGeneration != bundle.Generation {
```

Comment:
This check is not needed anymore because before adding new changes the values in `bundle.Status.ResourceKey` were populated by running helm template. This condition prevents running `setResourceKey` multiple times. We need this method to run everytime the bundle reconciler runs because `bundleDeployment.Status.Resources` update retriggers bundle reconciler

Reply:
I see, now that we rely on actual resources from bundle deployments, we need to update resource keys regardless of whether the bundle itself has changed. Thanks for clarifying.